tests/regression/task_spec.py

```diff
@@ -201,10 +217,19 @@ class SyntaxDiffSpec(TaskSpec):
     Extends TaskSpec by concrete syntax differences that should be found by
     DiffKemp. These are currently intended to be macros or inline assemblies.
     """
-    def __init__(self, spec, task_name, tasks_path, kernel_path):
-        TaskSpec.__init__(self, spec, task_name, tasks_path, kernel_path)
+    def __init__(self, spec, task_name, tasks_path, kernel_path,
+                 patterns_path):
+        TaskSpec.__init__(self, spec, task_name, tasks_path, kernel_path,
+                          patterns_path)
+        self.equal_diffs = dict()
```

Comment:
Is a different set really needed when the `DiffSpec` specifies the expected result? Moreover, couldn't you just set `def_old` and `def_new` to empty strings and get rid of the `DiffSpec.result` completely?

Reply:
If I understand correctly, then probably no, not necessary. However, I would argue that both dicts should stay separate. It seems really confusing to store "equal" results in a dict for "syntax_diffs".

Reply:
I agree, but this still feels like abusing `DiffSpec`. What about this:

1. Use a new field `equal_symbol` instead of `diff_symbol` in the YAML spec file.
2. Make `equal_diffs` a simple list of symbols.

Reply:
I will look into that!